### PR TITLE
Fix bug, bulk toggleable CheckboxList, with dynamic options

### PR DESCRIPTION
**Pull request by tanthammar**
If you have a CheckboxList 
- that is `bulkToggleable`
- and its `options` are dynamically updated based on another field value
- where the options may become `null`
- Alpine `x-show` and `select/deselect all` will break when the `options` are updated.

This PR fixes that.

Needed a lot of wire:keys and wrapping divs to get it to play nice, please don't think I am cluttering the component ...

### CODE REVIEW

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 1]
```diff
@@ -88,20 +89,23 @@
                             'dark:border-gray-600' => (! $errors->has($getStatePath())) && config('forms.dark_mode'),
                             'border-danger-300 ring-danger-500' => $errors->has($getStatePath()),
                             'dark:border-danger-400 dark:ring-danger-400' => $errors->has($getStatePath()) && config('forms.dark_mode'),
+                        ])->merge([
+                            'x-on:change' => $isBulkToggleable() ? 'updateIsAllSelected' : null
                         ]) }}
-                    />
+                        />
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php:96]: Could you fix those indentation changes? 🙂

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 2]
```diff
@@ -88,20 +89,23 @@
...
 
-                    <span @class([
+                        <span @class([
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php:98]: Here as well.

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/forms/resources/views/components/checkbox-list.blade.php (hunk 1 above)
Comment (tanthammar) [reply packages/forms/resources/views/components/checkbox-list.blade.php:96]: Not sure about this one, it is aligned with the start of the input tag, where it should be?

File: packages/forms/resources/views/components/checkbox-list.blade.php (hunk 2 above)
Comment (tanthammar) [reply packages/forms/resources/views/components/checkbox-list.blade.php:98]: I think I got it right now. Please review the indentions again.

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 3]
```diff
@@ -65,21 +72,15 @@
...
+            @forelse ($getOptions() as $optionValue => $optionLabel)
+                <div wire:key="{{ $getId() }}-{{ $getStatePath() }}-{{ $optionValue }}">
+                    <label class="flex items-center space-x-3 rtl:space-x-reverse">
+                        <input
+                            {!! $isDisabled() ? 'disabled' : null !!}
+                            wire:loading.attr="disabled"
+                            type="checkbox"
+                            value="{{ $optionValue }}"
+                            dusk="filament.forms.{{ $getStatePath() }}"
                         {{ $applyStateBindingModifiers('wire:model') }}="{{ $getStatePath() }}"
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: @tanthammar, here's where the indentation goes wrong.

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/forms/resources/views/components/checkbox-list.blade.php (hunk 3 above)
Comment (tanthammar) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: OKI :) ok now?

**Comment by tanthammar** [comment]
@zepfietje am I done? Looks like a pending change request but I can't see which one is pending ...

**Comment by zepfietje** [comment]
@tanthammar, I'll do a proper review when I find the time soon. 👍

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 4]
```diff
@@ -11,14 +11,16 @@
...
     <div x-data="{
         checkboxes: $root.querySelectorAll('input[type=checkbox]'),
 
         isAllSelected: false,
 
+        options: @js($getOptions()),
+
         init: function () {
             this.updateIsAllSelected()
+            $watch('options', this.updateIsAllSelected())
```
Comment (danharrin) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: /wtd add new line above

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 5]
```diff
@@ -34,26 +36,31 @@
...
-            <x-forms::link
-                tag="button"
-                size="sm"
-                x-show="!isAllSelected"
-                x-on:click="toggleAll"
-            >
-                {{ __('forms::components.checkbox_list.buttons.select_all.label') }}
-            </x-forms::link>
+        <div wire:key="{{ $getId() }}-toggleable-wrapper">
+            @if($isBulkToggleable() && count($getOptions()))
```
Comment (danharrin) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: /wtd fix spacing after `@if`

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 6]
```diff
@@ -34,26 +36,31 @@
...
+                <div  x-cloak class="mb-2" wire:key="{{ $getId() }}-toggleable-buttons">
```
Comment (danharrin) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: /wtd fix spacing after taf

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 7]
```diff
@@ -65,43 +72,39 @@
...
+                            {{ $applyStateBindingModifiers('wire:model') }}="{{ $getStatePath() }}"
+                            {{ $getExtraAttributeBag()->class([
+                                'text-primary-600 transition duration-75 rounded shadow-sm focus:border-primary-500 focus:ring-2 focus:ring-primary-500 disabled:opacity-70',
+                                'dark:bg-gray-700 dark:checked:bg-primary-500' => config('forms.dark_mode'),
+                                'border-gray-300' => ! $errors->has($getStatePath()),
+                                'dark:border-gray-600' => (! $errors->has($getStatePath())) && config('forms.dark_mode'),
+                                'border-danger-300 ring-danger-500' => $errors->has($getStatePath()),
+                                'dark:border-danger-400 dark:ring-danger-400' => $errors->has($getStatePath()) && config('forms.dark_mode'),
+                            ])->merge([
+                                'x-on:change' => $isBulkToggleable() ? 'updateIsAllSelected' : null
```
Comment (danharrin) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: /wtd add trailing comma

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 8]
```diff
@@ -34,26 +37,31 @@
...
-            <x-forms::link
-                tag="button"
-                size="sm"
-                x-show="!isAllSelected"
-                x-on:click="toggleAll"
-            >
-                {{ __('forms::components.checkbox_list.buttons.select_all.label') }}
-            </x-forms::link>
+        <div wire:key="{{ $getId() }}-toggleable-wrapper">
+            @if($isBulkToggleable() && count($getOptions()))
```
Comment (danharrin) [inline packages/forms/resources/views/components/checkbox-list.blade.php]:
```suggestion
            @if ($isBulkToggleable() && count($getOptions()))
```

**Comment by danharrin** [comment]
Thanks

**Comment by nessimabadi** [comment]
Hi, sorry to write on this merged PR but I'm on the latest version of the Forms package and I'm having the same problem described before the PR was made.
Does the fix still works?

**Pull request merged**